src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml

```diff
@@ -0,0 +1,310 @@
+          " call. Make sure to map the event in 'RudderStack Page Name/Category to Optimizely event mappings' setting");
+          mappedEvents;
+
+      - name: optimizelyEvents
+        template: |
+          const mappedEvents = $.outputs.preparePageOrScreenPayload.mappedEvents;
+          const availableEvents = $.outputs.dataFile.events;
+          const eventsFound = [];
+          let event = availableEvents.(){.key.toLowerCase() === mappedEvents[0].toLowerCase()} 
+          event.id ? eventsFound.push(event);
+
+
+          event = availableEvents.(){.key.toLowerCase() === mappedEvents[1].toLowerCase()};
+          event.id ? eventsFound.push(event);
+
+          $.assert(eventsFound.length > 0, "Mapped events [" + mappedEvents.join(", ") + "] are not present in the data file. Make sure events exist in Optimizely.");
+          eventsFound;
+
+      - name: events
+        template: |
+          const optimizelyEvents = $.outputs.preparePageOrScreenPayload.optimizelyEvents;
+          const timestamp = .message.().(
+            {{{{$.getGenericPaths("timestamp")}}}};
+          );
+          const events = [
+            {
+            "entity_id": optimizelyEvents[0].id,
+            "timestamp": $.formatTimeStamp(timestamp),
+            "key": optimizelyEvents[0].key,
+            "uuid": .message.messageId,
+            "tags": .message.properties
+            }
+          ];
+
+          optimizelyEvents.length > 1 ? events.push({
+            "entity_id": optimizelyEvents[1].id,
+            "timestamp": $.formatTimeStamp(timestamp),
+            "key": optimizelyEvents[1].key,
+            "uuid": $.generateUUID(),
+            "tags": .message.properties
+          });
+          events;
+
+      - name: visitors
+        template: |
+          const snapshots = [{
+            "decisions": [],
+            "events": $.outputs.preparePageOrScreenPayload.events
+          }];
+          const visitorId = .destination.Config.trackKnownUsers ? .message.userId : .message.anonymousId;
+          const visitors = [{
+            "visitor_id": visitorId,
+            "snapshots": snapshots
+          }];
+          const sessionId = .message.().(
+            {{{{$.getGenericPaths("sessionId")}}}};
+          );
+          const attributes = $.outputs.getAttributes;
+          sessionId ? visitors[0].session_id = String(sessionId);
+          attributes.length > 0 ? visitors[0].attributes = attributes
+          visitors;
+
+      - name: payload
+        template: |
+          {
+            ...$.outputs.commonPayload,
+            "visitors": $.outputs.preparePageOrScreenPayload.visitors,
+          };
+
+  - name: buildResponseForProcessTransformation
+    description: build response
+    template: |
+      const messageType = $.outputs.messageType;
+      const payload = (messageType === {{$.EventType.IDENTIFY}} ? $.outputs.prepareIdentifyPayload:
```

Comment:
const payload = $.outputs.prepareIdentifyPayload || $.outputs.prepareTrackPayload || $.outputs.preparePageOrScreenPayload
Reason: only one of them will be defined right?